### PR TITLE
add package_dir in setup.py  on dump

### LINKED ISSUE #257: setup.py converter does not handle the case where packages are in a subdirectory
Take for example the sources here: https://github.com/sdispater/clikit/

The clikit module is located at `src/clikit/`, not `clikit/`. Trying to bootstrap this package using dephell to generate setup.py, I get:

```
# -*- coding: utf-8 -*-

# DO NOT EDIT THIS FILE!
# This file has been autogenerated by dephell <3
# https://github.com/dephell/dephell

try:
    from setuptools import setup
except ImportError:
    from distutils.core import setup


import os.path

readme = ''
here = os.path.abspath(os.path.dirname(__file__))
readme_path = os.path.join(here, 'README.rst')
if os.path.exists(readme_path):
    with open(readme_path, 'rb') as stream:
        readme = stream.read().decode('utf8')


setup(
    long_description=readme,
    name='clikit',
    version='0.3.1',
    description='CliKit is a group of utilities to build beautiful and testable command line interfaces.',
    python_requires='!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,<4.0,>=2.7',
    project_urls={'repository': 'https://github.com/sdispater/clikit'},
    author='Sébastien Eustace',
    author_email='sebastien@eustace.io',
    license='MIT',
    keywords='packaging dependency poetry',
    packages=['clikit', 'clikit.adapter', 'clikit.api', 'clikit.api.application', 'clikit.api.args', 'clikit.api.args.format', 'clikit.api.command', 'clikit.api.config', 'clikit.api.event', 'clikit.api.formatter', 'clikit.api.io', 'clikit.api.resolver', 'clikit.args', 'clikit.args.inputs', 'clikit.config', 'clikit.formatter', 'clikit.handler', 'clikit.handler.help', 'clikit.io', 'clikit.io.input_stream', 'clikit.io.output_stream', 'clikit.resolver', 'clikit.ui', 'clikit.ui.alignment', 'clikit.ui.components', 'clikit.ui.help', 'clikit.ui.layout', 'clikit.ui.style', 'clikit.utils'],
    package_data={},
    install_requires=['enum34==1.*,>=1.1.0; python_version == "2.7.*" and python_version >= "2.7.0"', 'pastel==0.*,>=0.1.0', 'pylev==1.*,>=1.3.0', 'typing==3.*,>=3.6.0; python_version == "2.7.*" and python_version >= "2.7.0" or python_version == "3.4.*" and python_version >= "3.4.0"'],
    extras_require={'dev': ['pre-commit==1.*,>=1.12.0', 'pytest==4.*,>=4.0.0', 'pytest-cov==2.*,>=2.6.0', 'tox==3.*,>=3.5.0']},
)
```

The problem is all these packages=[] need to be located relative to a `package_dir={'': 'src'}`, which is not specified here. So, running setup.py build results in the error:

```
running build
running build_py
error: package directory 'clikit' does not exist
```

### PR DESCRIPTION
**Pull request by orsinium**
fix #257

### CODE REVIEW

**Comment by orsinium** [comment]
Resulting `setup.py` for [clikit](https://github.com/sdispater/clikit/):

```python
# -*- coding: utf-8 -*-

# DO NOT EDIT THIS FILE!
# This file has been autogenerated by dephell <3
# https://github.com/dephell/dephell

try:
    from setuptools import setup
except ImportError:
    from distutils.core import setup

import os.path

readme = ''
here = os.path.abspath(os.path.dirname(__file__))
readme_path = os.path.join(here, 'README.rst')
if os.path.exists(readme_path):
    with open(readme_path, 'rb') as stream:
        readme = stream.read().decode('utf8')

setup(
    long_description=readme,
    name='clikit',
    version='0.4.0',
    description='CliKit is a group of utilities to build beautiful and testable command line interfaces.',
    python_requires='!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,<4.0,>=2.7',
    project_urls={'repository': 'https://github.com/sdispater/clikit'},
    author='Sébastien Eustace',
    author_email='sebastien@eustace.io',
    license='MIT',
    keywords='packaging dependency poetry',
    packages=[
        'clikit', 'clikit.adapter', 'clikit.api', 'clikit.api.application',
        'clikit.api.args', 'clikit.api.args.format', 'clikit.api.command',
        'clikit.api.config', 'clikit.api.event', 'clikit.api.formatter',
        'clikit.api.io', 'clikit.api.resolver', 'clikit.args',
        'clikit.args.inputs', 'clikit.config', 'clikit.formatter',
        'clikit.handler', 'clikit.handler.help', 'clikit.io',
        'clikit.io.input_stream', 'clikit.io.output_stream', 'clikit.resolver',
        'clikit.ui', 'clikit.ui.alignment', 'clikit.ui.components',
        'clikit.ui.help', 'clikit.ui.layout', 'clikit.ui.style', 'clikit.utils'
    ],
    package_dir={'': 'src'},
    package_data={},
    install_requires=[
        'enum34==1.*,>=1.1.0; python_version == "2.7.*" and python_version >= "2.7.0"',
        'pastel==0.*,>=0.1.0', 'pylev==1.*,>=1.3.0',
        'typing==3.*,>=3.6.0; python_version == "2.7.*" and python_version >= "2.7.0" or python_version == "3.4.*" and python_version >= "3.4.0"'
    ],
    extras_require={
        'dev': [
            'pre-commit==1.*,>=1.12.0', 'pytest==4.*,>=4.0.0',
            'pytest-cov==2.*,>=2.6.0', 'tox==3.*,>=3.5.0'
        ]
    },
)
```

File: dephell/converters/setuppy.py [hunk 1]
```diff
@@ -208,6 +208,8 @@ def dumps(self, reqs, project: RootDependency, content=None) -> str:
 
         # packages, package_data
         content.append(('packages', sorted(str(p) for p in project.package.packages)))
+        if project.package.package_dir:
```
Comment (jayvdb) [inline dephell/converters/setuppy.py]: This is emitting `package_dir={'': ''},` unnecessarily...?

Comment (orsinium) [reply dephell/converters/setuppy.py]: Yes, this is just to keep output clearer.

**Comment by eli-schwartz** [comment]
I've tried to build a package which was generated with these changes but does *not* have a subdirectory: https://github.com/dephell/dephell_discover/pull/9#issuecomment-552922231

It doesn't work, because it errors out on:

```
running install_egg_info
error: error in 'egg_base' option: '' does not exist or is not a directory
```

@orsinium I'm not sure it makes sense to "keep output clearer" here, it seems to be a little bit worse than merely unnecessary.

**Comment by orsinium** [comment]
Oh wait, it returns `{'': ''}`, and this is truely value :scream:

**Pull request merged**